pkg/frontend/querymiddleware/codec_test.go

```diff
@@ -1613,3 +1613,11 @@ func TestPrometheusCodec_DecodeMultipleTimes(t *testing.T) {
 func newTestPrometheusCodec() Codec {
 	return NewPrometheusCodec(prometheus.NewPedanticRegistry(), 0*time.Minute, formatJSON)
 }
+
+func mustSucceed[T any](value T, err error) T {
+	if err != nil {
+		panic(err)
```

Comment:
To me panic in test implies that it's catching a programming error that should never happen in real life. I'd use require.NoError() or assert.NoError and pass `t`.

Reply:
I agree with you. It's a bit annoying having to pass t because can't be a wrapper anymore. The closest thing would be:

```
func mustNoError[T any](t *testing.T) func(value T, err error) T {
       return func(value T, err error) T {
               require.NoError(t, err)
               return value
       }
}
```

And then you can call it like this:
```
-                       requestB:      mustSucceed(rangeQuery.WithStartEnd(rangeQuery.GetStart()+5*time.Minute.Milliseconds(), rangeQuery.GetEnd()+5*time.Minute.Milliseconds())),
+                       requestB:      mustNoError[MetricsQueryRequest](t)(rangeQuery.WithStartEnd(rangeQuery.GetStart()+5*time.Minute.Milliseconds(), rangeQuery.GetEnd()+5*time.Minute.Milliseconds())),
```

I think this latter version makes code a bit more ugly. Given the panic affects only tests I will move on.